core/src/main/java/com/linecorp/armeria/common/HttpRequestBuilder.java

```diff
@@ -0,0 +1,417 @@
+     */
+    public HttpRequestBuilder get(String path) {
+        return method(HttpMethod.GET).path(path);
+    }
+
+    /**
+     * Shortcut to create a new {@link HttpRequestBuilder} with POST method and path.
+     */
+    public HttpRequestBuilder post(String path) {
+        return method(HttpMethod.POST).path(path);
+    }
+
+    /**
+     * Shortcut to create a new {@link HttpRequestBuilder} with PUT method and path.
+     */
+    public HttpRequestBuilder put(String path) {
+        return method(HttpMethod.PUT).path(path);
+    }
+
+    /**
+     * Shortcut to create a new {@link HttpRequestBuilder} with DELETE method and path.
+     */
+    public HttpRequestBuilder delete(String path) {
+        return method(HttpMethod.DELETE).path(path);
+    }
+
+    /**
+     * Sets the method for this request.
+     *
+     * @see HttpMethod
+     */
+    public HttpRequestBuilder method(HttpMethod method) {
+        requireNonNull(method, "method");
+        requestHeadersBuilder.method(method);
+        return this;
+    }
+
+    /**
+     * Sets the path for this request.
+     */
+    public HttpRequestBuilder path(String path) {
+        requireNonNull(path, "path");
+        this.path = path;
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType mediaType, CharSequence content) {
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        requestHeadersBuilder.contentType(mediaType);
+        this.content = HttpData.of(mediaType.charset(StandardCharsets.UTF_8), content);
+        this.publisher = null;
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType mediaType, String content) {
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(content, "content");
+        requestHeadersBuilder.contentType(mediaType);
+        this.content = HttpData.of(mediaType.charset(StandardCharsets.UTF_8), content);
+        this.publisher = null;
+        return this;
+    }
+
+    /**
+     * Sets the content for this request. The content of the request is formatted by
+     * {@link String#format(Locale, String, Object...)} with {@linkplain Locale#ENGLISH English locale}.
+     */
+    @FormatMethod
+    public HttpRequestBuilder content(MediaType mediaType, @FormatString String format, Object... content) {
+        requireNonNull(mediaType, "mediaType");
+        requireNonNull(format, "format");
+        requestHeadersBuilder.contentType(mediaType);
+        this.content = HttpData.of(mediaType.charset(StandardCharsets.UTF_8), format, content);
+        this.publisher = null;
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType mediaType, byte[] content) {
```

Comment:
Global comment: It would better to use `contentType` for a parameter name because its value is used as a `Content-Type` header.

```suggestion
    public HttpRequestBuilder content(MediaType contentType, byte[] content) {
```